src/deleted.psm1

```diff
@@ -0,0 +1,975 @@
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$User
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    if ($User.Id -as [int])
+    {
+        $User = $User.Id
+    }
+
+    $local:RelPath = "Deleted/Users"
+    $local:ErrMsgSuffix = " in deleted users"
+
+    if (-not ($User -as [int]))
+    {
+        try
+        {
+            $local:Users = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "$($local:RelPath)" `
+                                 -Parameters @{ filter = "Name ieq '$User' or DisplayName ieq '$User'"; fields = "Id" })
+        }
+        catch
+        {
+            Write-Verbose $_
+            Write-Verbose "Caught exception with ieq filter, trying with q parameter"
+            $local:Users = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "$($local:RelPath)" `
+                                 -Parameters @{ q = $User; fields = "Id" })
+        }
+        if (-not $local:Users)
+        {
+            throw "Unable to find user matching '$User'$($local:ErrMsgSuffix)"
+        }
+        if ($local:Users.Count -ne 1)
+        {
+            throw "Found $($local:Users.Count) users matching '$User'$($local:ErrMsgSuffix)"
+        }
+        $local:User[0].Id
+    }
+    else
+    {
+        # Make sure it actually exists
+        $local:User = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "$($local:RelPath)" `
+            -Parameters @{ filter = "Id eq $User"; fields = "Id" })
+        if (-not $local:User)
+        {
+            throw "Unable to find user matching '$User'$($local:ErrMsgSuffix)"
+        }
+        $User
+    }
+}
+
+<#
+.SYNOPSIS
+Get deleted users.
+
+.DESCRIPTION
+Returns a list of users that have been soft-deleted from Safeguard. These users can
+be restored with the Restore-SafeguardDeletedUser command.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER UserToGet
+An integer containing the ID of the deleted user to get or a string containing the name.
+
+.PARAMETER Fields
+An array of the deleted user property names to return.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Get-SafeguardDeletedUser -AccessToken $token -Appliance 10.5.32.54 -Insecure
+
+.EXAMPLE
+Get-SafeguardDeletedUser -Fields Id,Name,NetworkAddress
```

Comment:
A user object doesn't have Name and NetworkAddress properties. Maybe use Id, UserName and DisplayName for the example